lib/ansible/galaxy/dependency_resolution/dataclasses.py

```diff
@@ -578,10 +578,9 @@ def is_pinned(self):
 
         See https://github.com/ansible/ansible/pull/81606 for extra context.
         """
-        version_string = self.ver[0]
-        return version_string.isdigit() or not (
-            version_string == '*' or
-            version_string.startswith(('<', '>', '!='))
+        version_spec_start_char = self.ver.strip()[0]
+        return version_spec_start_char.isdigit() or not (
+            version_spec_start_char.startswith(('<', '>', '!', '*'))
```

Comment:
The fix for `!=` lgtm, but the added `strip()` is inconsistent with other codepaths. Something like `ansible-galaxy collection install "amazon.aws:= 1.0.0"` causes `ERROR! Non integer values in LooseVersion (' 1.0.0')`.

If we want to allow more leniency for whitespace in version requirements, I would be in favor of normalizing that in from_requirement_dict for simplicity.